samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -621,20 +596,47 @@ def _enrich_mapped_resources(
         "IMAGE_LAMBDA_FUNCTION": _enrich_image_lambda_function,
     }
 
-    for sam_metadata_resource in sam_metadata_resources:
-        resource_type = sam_metadata_resource.resource.get("values", {}).get("triggers", {}).get("resource_type")
-        sam_metadata_resource_address = sam_metadata_resource.resource.get("address")
-        enrichment_function = resources_types_enrichment_functions.get(resource_type)
-        if enrichment_function is None:
-            raise InvalidSamMetadataPropertiesException(
-                f"The resource type {resource_type} found in the sam metadata resource {sam_metadata_resource_address} "
-                f"is not a correct resource type. The resource type should be one of these values "
-                f"{resources_types_enrichment_functions.keys()}"
+    # create output directory if it doesn't exist
+    if not os.path.exists(output_directory_path):
+        os.makedirs(output_directory_path, exist_ok=True)
+
+    # copy script copy_terraform_built_artifacts.py into output directory
+    copy_terraform_built_artifacts_script_path = os.path.join(
+        os.path.dirname(__file__), "copy_terraform_built_artifacts"
+    )
+    shutil.copy(copy_terraform_built_artifacts_script_path, output_directory_path)
+
+    # check python is installed and get python command
+    python_command_name = _get_python_command_name()
+
+    # create makefile
+    makefilePath = os.path.join(output_directory_path, "Makefile")
+    with open(makefilePath, "a+") as makefile:
+        for sam_metadata_resource in sam_metadata_resources:
+
+            # enrich resource
+            resource_type = sam_metadata_resource.resource.get("values", {}).get("triggers", {}).get("resource_type")
+            sam_metadata_resource_address = sam_metadata_resource.resource.get("address")
+            enrichment_function = resources_types_enrichment_functions.get(resource_type)
+            if enrichment_function is None:
+                raise InvalidSamMetadataPropertiesException(
+                    f"The resource type {resource_type} found in the sam metadata resource "
+                    f"{sam_metadata_resource_address} is not a correct resource type. The resource type should be one "
+                    f"of these values {resources_types_enrichment_functions.keys()}"
+                )
+            cfn_resource, logical_id = _get_relevant_cfn_resource(sam_metadata_resource, cfn_resources)
+            enrichment_function(
```

Comment:
I'm not following this bit, is `resources_types_enrichment_functions` a dictionary of functions, where is it defined?

Reply:
the enrichment logic I didn't add, that was already there. I just put my changes and the enrichment stuff in the same function.
But yes, resources_types_enrichment_functions is a dictionary of functions, defined in this same function (a few lines above)